src/components/AvatarWithImagePicker.js

```diff
@@ -104,11 +112,37 @@ class AvatarWithImagePicker extends React.Component {
     }
 
     /**
-     *  Toggle max upload limit modal's visibility
-     * @param {Boolean} isVisible
-    */
-    setUploadLimitModalVisibility(isVisible) {
-        this.setState({isMaxUploadSizeModalOpen: isVisible});
+     *  Show the error modal
+     * @param {String} title
+     * @param {String} prompt
+     */
+    showErrorModal(title, prompt) {
+        this.setState({isErrorModalVisible: true, errorModalTitle: title, errorModalPrompt: prompt});
+    }
+
+    /**
+     *  Hide the error modal
+     */
+    hideErrorModal() {
+        this.setState({isErrorModalVisible: false});
+    }
+
+    /**
+     *  Checks if the image has the valid size and triggers the onImageSelected callback
+     * @param {Object} image
+     */
+    updateAvatarImage(image) {
+        if (this.isValidSize(image)) {
+            this.props.onImageSelected(image);
+            return;
+        }
+
+        // Since in react native we can't show two modals simultaneously
+        // we have to add delay before opening another one
+        setTimeout(() => this.showErrorModal(
```

Comment:
What is the other modal here? How can I test this?

Reply:
AvatarCropModal.
Steps to reproduce: 

1. Replace `updateAvatarImage`'s code with:
```javascript
    updateAvatarImage(image) {
        if (this.isValidSize(image) && false) {
            this.props.onImageSelected(image);
            return;
        }

        // Since in react native we can't show two modals simultaneously
        // we have to add delay before opening another one
        setTimeout(() => this.showErrorModal(
            this.props.translate('avatarWithImagePicker.imageUploadFailed'),
            this.props.translate('avatarWithImagePicker.sizeExceeded', {maxUploadSizeInMB: CONST.AVATAR_MAX_ATTACHMENT_SIZE / (1024 * 1024)}),
        ), 0);
    }
 ```

2. Then open the AvatarCropModal by selecting any image
3. Press the Save button 
4. Try to open the AvatarCropModal again

It will be impossible to open another modal, because of https://github.com/react-native-modal/react-native-modal/issues/30